Lib/fontTools/ttLib/tables/O_S_2f_2.py

```diff
@@ -508,11 +506,20 @@ def intersectUnicodeRanges(unicodes, inverse=False):
 	True
 	"""
 	unicodes = set(unicodes)
-	uniranges = _getUnicodeRangeSets()
-	bits = set([
-		bit for bit, unirange in enumerate(uniranges)
-		if not unirange.isdisjoint(unicodes) ^ inverse])
-	return bits
+	unicodestarts, unicodevalues = _getUnicodeRanges()
+	bits = set()
+	for code in unicodes:
+		stop, bit = unicodevalues[bisect.bisect(unicodestarts, code)]
+		if code <= stop:
+			bits.add(bit)
+	for code in unicodes:
+		# The spec says that bit 57 ("Non Plane 0") implies that there's
+		# at least one codepoint beyond the BMP; so I also include all
+		# the non-BMP codepoints here
+		if 0x10000 <= code < 0x110000:
+			bits.add(57)
+			break
```

Comment:
Suggestion:
```python
# The spec says ... etc.
if any(0x10000 <= code < 0x110000 for code in unicodes):
    bits.add(57)
```

Reply:
But actually, there's no reason to do this test during lookup at all: non-BMP bit 57 should be incorporated into the `_unicodeStarts` and `_unicodeValues` lists.

Reply:
> Suggestion: […]

Of course…

Reply:
> But actually, there's no reason to do this test during lookup at all: non-BMP bit 57 should be incorporated into the `_unicodeStarts` and `_unicodeValues` lists.

See above.